lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -34,16 +37,38 @@
   private final int megaBytes;
   private final int number;
   private final int cores;
+  @Deprecated
   private final ResourceCatalog.Descriptor descriptor;
+  private final List<String> nodeNames;
+  private final List<String> rackNames;
 
+  /**
+   * Deprecated constructor. ResourceCatalog.Descriptor should not be used
+   * anymore. In order to specify the rack names and node names where evaluators
+   * should run on, you should use nodeNames and rackNames lists
+   *
+   * @deprecated since 0.12.0-incubating-SNAPSHOT, use constructor with node
```

Comment:
Just use "0.12", as that will be the release in which it is deprecated. SNAPSHOT isn't a release. And `incubating` will at some point dissappear from the version string.